sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsVacuum.java

```diff
@@ -0,0 +1,18 @@
+package org.apache.spark.sql.connector.catalog;
+
+import org.apache.spark.annotation.Experimental;
+/**
+ * A mix-in interface for {@link Table} vacuum support. Data sources can implement this
+ * interface to provide the ability to perform table maintenance on request of the user.
+ */
+@Experimental
+public interface SupportsVacuum {
+
+  /**
+   * Performs maintenance on the table.  This often includes removing unneeded data and
```

Comment:
Should this be "unneeded metadata" instead?

I think that the behavior should be that vacuum does not modify the table's contents. DML operations should be used for that. Vacuum only cleans up metadata for faster operations, like an OPTIMIZE.